### DISCUSSION TITLE
片方のキーボードが時間経過で反応しなくなる

### DISCUSSION THREAD

**Discussion by work-hadano** [Yowkees/keyball#719] [Q&A] [answered]
質問失礼します。
Keyball44の組み立てが終わり、左右のキーボードを繋げてPCに接続したところ、
以下の問題が発生しました。お手数ですが、気づくことがありましたら、アドバイス等いただけると幸いです。

【症状】
- 左右のキーボードをTRRSケーブル（4極）で繋げてPCに接続すると最初は左右共入力が出来るが時間が経過すると、左側のキーボードのLEDが黄色に変わり、反応しなくなる（正常に反応する時LEDは赤）（反応しなくなる時間はまちまちです。約30秒～3分）

【前提条件】
- 左右のキーボードに正規ファームウェア書き込み済み
- TRRSケーブルを繋げずに片方のキーボード単体でUSBに繋げば入力可能

【試した事】
- TRRSジャックのはんだ付けし直し
- TRRSケーブルの交換

正常に反応している時、反応しなくなった時、裏面のはんだづけの写真を添付いたします。
お手数おかけしますが、宜しくお願いいたします。

![S__49233927_0](https://github.com/user-attachments/assets/222c7d2b-892e-47b2-a6aa-437c7477b9a7)
![S__49233929_0](https://github.com/user-attachments/assets/cb46462d-cd78-41d2-803a-3d5e31a3284c)
![S__49233930_0](https://github.com/user-attachments/assets/5800532c-c281-44f6-8988-6800e4483bfb)
![S__49233931_0](https://github.com/user-attachments/assets/8e5ebd26-68e8-4b3a-9fb4-968e337038b8)

**Comment by work-hadano** [comment]
追記
時間経過後、反応しなくなった状態でTRRSケーブルを触ったりいじったりしていると、反応するようになったりします。
2本新品でTRRSケーブルを買って試していますが、反応しなくなる頻度は違えど、どちらも不安定なところがあります。
新品なのでケーブル自体が原因ではなさそうだと考えていますが、他に関係している箇所などありますでしょうか？

**Comment by Yowkees** [comment, marked as answer]
ご連絡ありがとうございます。

左右の通信が不安定になっているようで、多くの場合TRSジャックの半田付けが原因だったりしますが、お写真ではキレイに半田付けされているように思います。
私が使用しているKeyballでも、TRSコネクタが回転したりすると左右の通信が途切れることがあります。
そのため、屈曲性が良く軽いケーブルを使用した方が安定するかと思います。

また、フルカラーLEDを明るく白っぽい色にすると電流が足りずにProMicroが落ちることもあります。
これらにご注意いただいて様子をみていただき、何かございましたらお知らせください。

**Comment by work-hadano** [comment]
> また、フルカラーLEDを明るく白っぽい色にすると電流が足りずにProMicroが落ちることもあります。

色々と試してみたところ、どうやら上記の通りLEDが関係しているようです。
LEDをOFFにすると安定して使用できるようになりました。アドバイスありがとうございました。

**Comment by Yowkees** [comment]
ご確認ありがとうございます。LED色の工夫で解決できたようで良かったです。安心しました。
Keyball44はLEDが多いのでProMicroの限界に達しやすいかも知れません。
ぜひ末永くKeyball44をご愛用いただければ幸いです。
